src/Sarif.Viewer.VisualStudio/Models/SarifErrorListItem.cs

```diff
@@ -39,13 +38,14 @@ internal SarifErrorListItem()
             _fixes = new ObservableCollection<FixModel>();
         }
 
-        public SarifErrorListItem(Run run, Result result, string logFilePath) : this()
+        public SarifErrorListItem(Run run, Result result, string logFilePath, ProjectNameCache projectNameCache) : this()
         {
             IRule rule;
             run.TryGetRule(result.RuleId, result.RuleKey, out rule);
             Message = result.GetMessageText(rule, concise: false);
             ShortMessage = result.GetMessageText(rule, concise: true);
             FileName = result.GetPrimaryTargetFile();
+            ProjectName = projectNameCache.GetName(FileName);
```

Comment:
>ProjectName [](start = 12, length = 11)

Populate the new `ProjectName` property from the cache each time we create a `SarifErrorListItem`. Now the cache need not persist once all the error list items have been created.

Reply:
Having this contained here inside the SarifErrorListItem makes a lot more sense.  Setting the ProjectName is no longer the parents responsibility.  

---
In reply to: [135426373](https://github.com/Microsoft/sarif-sdk/pull/599#discussion_r135426373) [](ancestors = 135426373)